tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -1275,24 +1275,40 @@ def test_update_run_name(self):
 
         run_id = self.store.create_run(**configs).info.run_id
         run = self.store.get_run(run_id)
-        self.assertEqual(run.info.run_name, configs["run_name"])
+        self.assertEqual(configs["run_name"], run.info.run_name)
 
         self.store.update_run_info(run_id, RunStatus.FINISHED, 1000, "new name")
         run = self.store.get_run(run_id)
-        self.assertEqual(run.info.run_name, "new name")
-        self.assertEqual(run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME), "new name")
+        self.assertEqual("new name", run.info.run_name)
+        self.assertEqual("new name", run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME))
 
         self.store.update_run_info(run_id, RunStatus.FINISHED, 1000, None)
         run = self.store.get_run(run_id)
-        self.assertEqual(run.info.run_name, "new name")
-        self.assertEqual(run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME), "new name")
+        self.assertEqual("new name", run.info.run_name)
+        self.assertEqual("new name", run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME))
 
         self.store.delete_tag(run_id, mlflow_tags.MLFLOW_RUN_NAME)
         run = self.store.get_run(run_id)
-        self.assertEqual(run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME), None)
+        self.assertEqual(None, run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME))
+
         self.store.update_run_info(run_id, RunStatus.FINISHED, 1000, "newer name")
         run = self.store.get_run(run_id)
-        self.assertEqual(run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME), "newer name")
+        self.assertEqual("newer name", run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME))
+
+        self.store.set_tag(run_id, entities.RunTag(mlflow_tags.MLFLOW_RUN_NAME, "newest name"))
+        run = self.store.get_run(run_id)
+        self.assertEqual(run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME), "newest name")
+        self.assertEqual("newest name", run.info.run_name)
+
+        self.store.log_batch(
+            run_id,
+            metrics=[],
+            params=[],
+            tags=[entities.RunTag(mlflow_tags.MLFLOW_RUN_NAME, "batch name")],
+        )
+        run = self.store.get_run(run_id)
+        self.assertEqual(run.data.tags.get(mlflow_tags.MLFLOW_RUN_NAME), "batch name")
+        self.assertEqual("batch name", run.info.run_name)
```

Comment:
I switched these from the old test since when I was debugging, the values were swapped. The first arg prints out the expected value and the 2nd arg will print the actual. 

```python
    def test_update_run_name(self):
        fs = FileStore(self.test_root)
        run = self._create_run(fs)
        run_id = run.info.run_id

        self.assertEqual("name", run.info.run_name)
        self.assertEqual("intentionally wrong", run.data.tags.get(MLFLOW_RUN_NAME))
```

```bash
============================= test session starts ==============================
collecting ... collected 1 item

test_file_store.py::TestFileStore::test_update_run_name FAILED           [100%]
tests/store/tracking/test_file_store.py:1493 (TestFileStore.test_update_run_name)
'name' != 'intentionally wrong'

Expected :'intentionally wrong'
Actual   :'name'

```

Reply:
@BenWilson2 Looks like this is a PyCharm feature:

```
pytest tests/store/tracking/test_sqlalchemy_store.py -k test_update_run_name
```

gives:

```
    def test_update_run_name(self):
        experiment_id = self._experiment_factory("test_update_run_name")
        configs = self._get_run_configs(experiment_id=experiment_id)
    
        run_id = self.store.create_run(**configs).info.run_id
        run = self.store.get_run(run_id)
        self.assertEqual(configs["run_name"], run.info.run_name)
    
        self.store.update_run_info(run_id, RunStatus.FINISHED, 1000, "new name")
        run = self.store.get_run(run_id)
>       self.assertEqual("new name", "foo")
E       AssertionError: 'new name' != 'foo'
E       - new name
E       + foo
```

Reply:
- https://youtrack.jetbrains.com/issue/PY-27267
- https://youtrack.jetbrains.com/issue/PY-26471/Unittest-only-AssertEqual-assumes-that-the-first-param-is-the-expected-result-second-is-the-actual

Reply:
ahhh. Interesting that there's a wrapper in there around pytest functionality. I'll file a change PR :)